service/src/main/java/com/ericsson/eiffel/remrem/generate/controller/RemremGenerateController.java

```diff
@@ -89,11 +111,62 @@
                 return new ResponseEntity<>(parser.parse(RemremGenerateServiceConstants.NO_SERVICE_ERROR),
                         HttpStatus.SERVICE_UNAVAILABLE);
             }
+        } catch (REMGenerateException e1) {
+            if (e1.getMessage().contains(Integer.toString(HttpStatus.NOT_ACCEPTABLE.value()))) {
+                return new ResponseEntity<>(parser.parse(e1.getMessage()), HttpStatus.NOT_ACCEPTABLE);
+            } else {
+                return new ResponseEntity<>(parser.parse(e1.getMessage()), HttpStatus.EXPECTATION_FAILED);
+            }
         } catch (Exception e) {
-            return new ResponseEntity<>(parser.parse(RemremGenerateServiceConstants.INTERNAL_SERVER_ERROR), HttpStatus.INTERNAL_SERVER_ERROR);
+            return new ResponseEntity<>(parser.parse(RemremGenerateServiceConstants.INTERNAL_SERVER_ERROR),
+                    HttpStatus.INTERNAL_SERVER_ERROR);
         }
     }
 
+    private JsonObject erLookup(final JsonObject bodyJson, Boolean failIfMultipleFound, Boolean failIfNoneFound)
+            throws REMGenerateException {
+
+        // Checking ER lookup enabled or not
+        if (erlookupConfig.getEventRepositoryEnabled() && bodyJson.toString().contains("%lookup%")) {
+            JsonArray lookupLinks = bodyJson.get("eventParams").getAsJsonObject().get("links").getAsJsonArray();
+            JsonArray links = new JsonArray();
+            for (int i = 0; i < lookupLinks.size(); i++) {
+                if (lookupLinks.get(i).toString().contains("%lookup%")) {
+                    String[] ids = null;
+
+                    // prepare ER Query
+                    String Query = ERLookupController.getQueryfromLookup(lookupLinks.get(i).getAsJsonObject());
+                    String url = erlookupConfig.getErURL() + Query;
+
+                    // Execute ER Query
+                    response = restTemplate.getForEntity(url, String.class);
+                    if (response.getStatusCode() == HttpStatus.OK) {
+                        System.out.println("The result from Event Repository is: " + response.getStatusCodeValue());
+                        String responseBody = response.getBody();
+                        ids = ERLookupController.getIdsfromResponseBody(responseBody);
```

Comment:
should add retry when the ER lookup fails with 500/503 status code. Retry once for the intial patchset.